homeassistant/config_entries.py

```diff
@@ -331,6 +331,17 @@ async def setup_again(*_: Any) -> None:
         else:
             self.state = ENTRY_STATE_SETUP_ERROR
 
+    async def async_shutdown(self) -> None:
```

Comment:
Why should this be a coroutine function? We don't await inside.

Reply:
I made it a coroutine because I figured we would need to add more to it later (https://github.com/home-assistant/core/pull/49241) and it would end up being a coroutine anyways which would be a breaking change in the api I wanted to avoid in the future.

Reply:
Sounds good!